src/rp2_common/hardware_pwm/include/hardware/pwm.h

```diff
@@ -144,7 +144,10 @@ static inline void pwm_config_set_clkdiv_int(pwm_config *c, uint div) {
  * high level, rising edge or falling edge of the B pin input.
  */
 static inline void pwm_config_set_clkdiv_mode(pwm_config *c, enum pwm_clkdiv_mode mode) {
-    valid_params_if(PWM, mode >= PWM_DIV_FREE_RUNNING && mode <= PWM_DIV_B_FALLING);
+    valid_params_if(PWM, mode == PWM_DIV_FREE_RUNNING ||
+            mode == PWM_DIV_B_RISING ||
+            mode == PWM_DIV_B_HIGH ||
+            mode == PWM_DIV_B_FALLING);
```

Comment:
a) it probably makes sense to list these in the same order they're listed in the enum? (i.e. swap round `PWM_DIV_B_RISING` and `PWM_DIV_B_HIGH`)

b) as this is a multi-line chunk of code repeated in both `pwm_config_set_clkdiv_mode` and `pwm_set_clkdiv_mode` it's probably worth factoring it out to an(other) inline function?

Reply:
a) don't care enough to change it.
b) thought about it at the time, but decided against because it is non functional code... and i was just working around a compiler limitation